src/lib.rs

```diff
@@ -90,50 +99,186 @@ pub struct BigText<'a> {
+/// Returns how many cells are needed to display a full 8x8 glyphe using the given font size
+fn cells_per_glyph(size: &BigTextSize) -> (u16, u16) {
+    match size {
+        BigTextSize::Full => (8, 8),
+        BigTextSize::HalfHeight => (8, 4),
+        BigTextSize::HalfWidth => (4, 8),
+        BigTextSize::Half => (4, 4),
+    }
+}
+
+/// Chunk the area into as many x*y cells as possible returned as a 2D iterator of `Rect`s
 /// representing the rows of cells.
-fn layout(area: Rect) -> impl IntoIterator<Item = impl IntoIterator<Item = Rect>> {
-    (area.top()..area.bottom()).step_by(8).map(move |y| {
-        (area.left()..area.right()).step_by(8).map(move |x| {
-            let width = min(area.right() - x, 8);
-            let height = min(area.bottom() - y, 8);
-            Rect::new(x, y, width, height)
+/// The size of each cell depends on given font size
+fn layout(
+    area: Rect,
+    font_size: &BigTextSize,
+) -> impl IntoIterator<Item = impl IntoIterator<Item = Rect>> {
+    let (width, height) = cells_per_glyph(font_size);
+    (area.top()..area.bottom())
+        .step_by(height as usize)
+        .map(move |y| {
+            (area.left()..area.right())
+                .step_by(width as usize)
+                .map(move |x| {
+                    let width = min(area.right() - x, width);
+                    let height = min(area.bottom() - y, height);
+                    Rect::new(x, y, width, height)
+                })
         })
-    })
 }
 
 /// Render a single grapheme into a cell by looking up the corresponding 8x8 bitmap in the
 /// `BITMAPS` array and setting the corresponding cells in the buffer.
-fn render_symbol(grapheme: StyledGrapheme, area: Rect, buf: &mut Buffer) {
+fn render_symbol(grapheme: StyledGrapheme, area: Rect, buf: &mut Buffer, font_size: &BigTextSize) {
     buf.set_style(area, grapheme.style);
     let c = grapheme.symbol.chars().next().unwrap(); // TODO: handle multi-char graphemes
     if let Some(glyph) = font8x8::BASIC_FONTS.get(c) {
-        render_glyph(glyph, area, buf);
+        render_glyph(glyph, area, buf, font_size);
+    }
+}
+
+/// Get the correct unicode symbol for two vertical "pixels"
+fn get_symbol_half_height(top: u8, bottom: u8) -> &'static str {
+    match top {
+        0 => match bottom {
+            0 => " ",
+            _ => "▄",
+        },
+        _ => match bottom {
+            0 => "▀",
+            _ => "█",
+        },
+    }
+}
+
+/// Get the correct unicode symbol for two horizontal "pixels"
+fn get_symbol_half_width(left: u8, right: u8) -> &'static str {
+    match left {
+        0 => match right {
+            0 => " ",
+            _ => "▐",
+        },
+        _ => match right {
+            0 => "▌",
+            _ => "█",
+        },
+    }
+}
+
+/// Get the correct unicode symbol for 2x2 "pixels"
+fn get_symbol_half_size(
+    top_left: u8,
+    top_right: u8,
+    bottom_left: u8,
+    bottom_right: u8,
+) -> &'static str {
+    match top_left {
```

Comment:
I suspect this function would probably read better if represented as a 2D array rather than as match statements.
Something like (untested):
```rust
const chars: [[char; 4]; 4] = [
  [" ", "▘", "▝", "▀"],
  ["▖", "▌", "▞", "▛"],
  ["▗", "▚", "▐", "▜"],
  ["▄", "▙", "▟", "█"],
];
chars[bottom_left * 2 + bottom_right][top_left*2 + top_right]
```
If you look at the layout of the legacy symbols for quadrants, they're actually conveniently in this order, so you could probably just do something like (psuedo code)
```rust
let offset = bottom_left* 8 + bottom_right * 4 + top_left*2 + top_right;
if offset = 0 { " " } else { /* add offset to "\u{1FB00}" */ }
```

Reply:
I was somehow not able to get the 2D array running and keep the code pretty, but I now use a normal 1D array and index it using the offset.

Following your second  suggestion somehow also didn't work out as fast, so I kept the first array solution for now.

Anyway, as I searched for the unicode you mentioned "1FB00" and found out it is "Block Sextant". That is not the quadrants I used, but might be useful to implement another size later (with the current naming it would be something like "HalfWidthThirdHight"). But for now I would like to get this pull request polished and merged.